mlflow/pipelines/artifacts.py

```diff
@@ -0,0 +1,149 @@
+import json
+import logging
+import os
+
+import mlflow
+from mlflow.pipelines.utils.execution import get_step_output_path
+from mlflow.tracking import MlflowClient
+from mlflow.tracking._tracking_service.utils import _use_tracking_uri
+
+_logger = logging.getLogger(__name__)
+
+
+class DataframeArtifact:
```

Comment:
Should we make an abstract `Artifact` class object? This way we can enforce that `load()` is implemented.

Reply:
Sure thing - @jinzhang21 this overlaps a tad with your feedback, is this a reasonable first step here? This avoids increasing the scope of the refactor, and we can unify the various artifact classes in the mlflow codebase separately, if desired?

Reply:
We can merge your change as is w/o doing the extra step. But I want to get us aligned on the overall directions and the "big picture" here, namely, do we want Artifact class to exist in MLflow in the near future, what's the pros and cons of having it etc. Maybe a small design doc clarifying this decision is a better place for discussion. No need to do everything in this PR.